pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java

```diff
@@ -1046,6 +1047,41 @@ public void triggerOffload(
         internalTriggerOffload(authoritative, messageId);
     }
 
+    @PUT
+    @Path("/{tenant}/{namespace}/{topic}/partitioned-offload")
```

Comment:
Can we use `offload`? If the topic is a partitioned topic, offload all partitions.

Reply:
I've actually tried to use `offload` whose parameter is `MessageId`, however there seems to be no way to cast `MessageIdImpl` back to an object that contains `Map<partition,MessageIdImpl>` at the moment. Maybe we can get rid of parameter `MessageId` and obtain it within `PersistentTopicsBase.internalTriggerOffload` but it is an API breaking change.

Reply:
Do we need to reconsider offload by a message ID on a partitioned topic? If users need to specify a Map<partition, messageId>, this does not bring convenience,  In my opinion, this is more like a bulk API. If a bulk operates,  topics can across multiple partitioned topics.

Reply:
Do you mean all partitions using the same `MessageId`? I am not so sure about how to handle the required parameter `sizeThresholdStr` that is used to find `MessageId` in a non-partitioned topic (via the result of `PersistentTopicInternalStats`). In my opinion, multiple partitioned topics may have various `messageId` since the amount of data of each partition are different.

Reply:
No, I mean do we need to support offload by message ID for the partitioned topic? Reset cursor just supports reset by the timestamp on partitioned topic and reset by message ID on a partition.

Reply:
Honestly I'm not really sure, according to the current usage of `offload` admin cli it seems to be useful since this allows to set size thresholds for a topic. Maybe I misunderstood the original issue, what do you suggest?